src/main/java/io/confluent/examples/streams/microservices/util/MicroserviceUtils.java

```diff
@@ -58,8 +60,11 @@ public static Properties baseStreamsConfig(String bootstrapServers, String state
     config.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
     config.put(StreamsConfig.STATE_DIR_CONFIG, stateDir);
     config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
-    config.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, "exactly_once");
+    // Commenting out configuration for Exactly Once Semantics (EOS)
+    // because currently EOS is not supported with Confluent Monitoring Interceptors
+    //config.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, "exactly_once");
```

Comment:
this thought just occurred to me, instead of commenting out the config we could do something like this:
```java
public static Properties baseStreamsConfigEOS(final String bootstrapServers,
                                                final String stateDir,
                                                final String appId) {
    return baseStreamsConfig(bootstrapServers, stateDir, appId, true);
  }

  public static Properties  baseStreamsConfig(final String bootstrapServers,
                                              final String stateDir,
                                              final String appId) {
    return baseStreamsConfig(bootstrapServers, stateDir, appId, false);
  }
  
  public static Properties baseStreamsConfig(final String bootstrapServers,
                                             final String stateDir,
                                             final String appId,
                                             final boolean enableEOS) {
    Properties config = new Properties();
    // Workaround for a known issue with RocksDB in environments where you have only 1 cpu core.
    config.put(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG, CustomRocksDBConfig.class);
    config.put(StreamsConfig.APPLICATION_ID_CONFIG, appId);
    config.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
    config.put(StreamsConfig.STATE_DIR_CONFIG, stateDir);
    config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
    final String eosConfig = enableEOS ? "exactly_once" : "at_least_once";
    config.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, eosConfig);
    config.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 1); //commit as fast as possible

    return config;
  }
```
That way you can turn on EOS where ever you want, and nothing has to change in existing code.
WDYT?

Reply:
I like it!!